lib/options.go

```diff
@@ -139,29 +139,44 @@ type TLSAuthFields struct {
 
 // Defines a TLS client certificate to present to certain hosts.
 type TLSAuth struct {
-	TLSAuthFields
-	certificate *tls.Certificate
+	Origin            []*TLSAuthFields
+	Certificates      []tls.Certificate
+	NameToCertificate map[string]*tls.Certificate
```

Comment:
```suggestion
	certificates      []tls.Certificate
```
I think we should unexport and use them as a cache like it was done before, we can change the Certificate method for returning the slice instead of the single Certificate.
Also, I think it's better to maintain the logic for `NameToCertificate` close to the deprecation warning, so we will change only one place when we will rid the support.